go/ephemeral/common.go

```diff
@@ -65,6 +65,18 @@ func ctimeIsStale(ctime keybase1.Time, currentMerkleRoot libkb.MerkleRoot) bool
 	return currentMerkleRoot.Ctime()-ctime.UnixSeconds() >= KeyLifetimeSecs
 }
 
+// If a teamEK is almost expired we allow it to be created in the background so
+// content generation is not blocked by key generation. We *cannot* create a
+// teamEK in the background if the key is expired however since the current
+// teamEK's lifetime (and supporting device/user EKs) is less than the maximum
+// lifetime of ephemeral content. This can result in content loss once the keys
+// are deleted.
+func backgroundKeygenPossible(ctime time.Time, currentMerkleRoot libkb.MerkleRoot) bool {
+	diff := keybase1.TimeFromSeconds(currentMerkleRoot.Ctime()).Time().Sub(ctime)
+	keygenInterval := time.Second * time.Duration(KeyGenLifetimeSecs)
+	return diff >= (keygenInterval-time.Hour) && diff < keygenInterval
```

Comment:
This would be easier for me to follow with some more concrete variable names. Maybe `keyAge` or something instead of `diff`? And maybe give the return value an intermediate name like `isOneHourFromExpiration`?

Reply:
i'll change this in the subsequent PR with deletion fixes